support/gitignore

```diff
@@ -39,3 +39,26 @@ CPPLINT.cfg
 .deps/
 .libs/
 *-stamp
+
+.clang-format
```

Comment:
Several of those files already appear above, e.g. `clang-format`, `Makefile.in`, etc - could you make sure there are no duplicates?

Also, it would be nice to add them under the relevant sections.